src/sidebar/store/modules/annotations.js

```diff
@@ -260,6 +249,7 @@ function addAnnotations(annotations, now) {
     dispatch({
       type: actions.ADD_ANNOTATIONS,
       annotations: annotations,
+      currentAnnotationCount: getState().base.annotations.length,
```

Comment:
So to summarize the pattern here: Where existing reducers rely state from multiple modules, we're either:

- Moving the logic to a different reducer so that each reducer only depends on its own state. This is the preferred approach.
- If that's not possible, instead calculating the required information in the action creator, which may involve turning it into a thunk, and adding that as information to the action as part of the payload. The downside is that e.g. `ADD_ANNOTATIONS` needs to know about the requirements of the selection module and includes additional information beyond that needed to describe what happened. The upside is that it allows the "selection" module reducers to avoid needing to access external state.

I think it would be interesting to know whether the wider Redux community have found alternative patterns/approaches for this problem that could work well.